### ISSUE TITLE
Unable to Save PDF with Edits

### ISSUE THREAD

**Issue by WB-Student** [pdfarranger/pdfarranger#1019]
Describe the bug
----------------

Hello, I was unable to save any edits made to certain PDFs in PDFarranger today. I was able to complete my edits in the application (e.g., rotate a single page, duplicate a page) however, when attempting to save this edited PDF, I would receive a pop-up notification warning: "Forms and outlines are lost on saving." This in itself has not been a problem for me. However, this was followed by another pop-up notification with a series of letters and numbers (see two examples below) and only the option to click "Ok" of close that window; neither of which led to saving the edited file.  

So far, the only work around I have found is to "print" the target PDF to a new PDF. Then I can successfully edit and save that new "printed" file. 

To Reproduce
------------

1.  Go to the target PDF file in a file manager.
2.  Right click on target PDF file, select to open with PDFarranger. 
4.  Edit the file (e.g., rotate a page) and click Save or Save As.
5.  Receive the pop-up notifications and cannot save edited file. 

Expected behavior
-----------------

Successfully save the PDF file with edits retained.

Input files
-----------

Only appears to affect _some_ PDF files. I cannot attach an example here because they are copyrighted academic journal articles. However, I confirmed that the PDF files that cannot be saved after editing are PDF v. 1.7 and the PDF files that _can_ be saved after editing are PDF v. 1.3 if that is useful.

Screenshots
-----------

![Screenshot from 2023-12-26 14-16-57](https://github.com/pdfarranger/pdfarranger/assets/34949247/793c85af-c33b-48d4-8c09-8357af4a7226)
![Screenshot from 2023-12-26 14-38-04](https://github.com/pdfarranger/pdfarranger/assets/34949247/9eb6e5e4-221c-4f2f-aee5-aff4820db3b3)
![Screenshot from 2023-12-26 14-04-04](https://github.com/pdfarranger/pdfarranger/assets/34949247/075de0a5-dda1-4dc6-9c71-16e4551ecb6f)
![Screenshot from 2023-12-26 14-15-25](https://github.com/pdfarranger/pdfarranger/assets/34949247/7baa5158-3b57-41a4-9acd-d3d7c392769a)

System and Versions
-------------------

-   PDF Arranger 1.10.1 (flatpak)
-   pikepdf 8.10.1 (installed with PDF Arranger flatpak)
-   Linux Mint 21.2 Cinnamon. 

I greatly appreciate any and all help. 

Thank you.

**Comment by dreua** [comment]
Could you try running it from command line and see if it prints any more useful messages?

Would you be comfortable sharing the PDF privately with me or other developers? My mail should be visible on my profile page.

**Comment by WB-Student** [comment]
Hello @dreua,

Thank you for your swift reply and offer to help. I have emailed you some of the PDF files I had trouble with today and one that did not give me any trouble for comparison.  

Also, for the benefit of others following along and echoed in my private email to you, I am sorry I do not currently know how to run PDF Arranger in the command line. 

I look forward to your (and others') thoughts.

Thank you.

**Comment by dreua** [comment]
This also happens when trying to open the document properties:

```
/usr/lib64/python3.11/site-packages/pikepdf/models/metadata.py:412: UserWarning: The metadata field /CrossmarkDomainExclusive with value 'pikepdf.String("true")' has no XMP equivalent, so it was discarded
  warn(msg)
Traceback (most recent call last):
  File "/home/da/git/pdfarranger/.venv/lib64/python3.11/site-packages/pdfarranger/pdfarranger.py", line 2537, in edit_metadata
    if metadata.edit(self.metadata, files, self.window):
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/da/git/pdfarranger/.venv/lib64/python3.11/site-packages/pdfarranger/metadata.py", line 208, in edit
    mergedmetadata = merge(metadata, pdffiles)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/da/git/pdfarranger/.venv/lib64/python3.11/site-packages/pdfarranger/metadata.py", line 110, in merge
    return merge_doc(metadata, docs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/da/git/pdfarranger/.venv/lib64/python3.11/site-packages/pdfarranger/metadata.py", line 98, in merge_doc
    for k, v in _safeiter(meta.items()):
  File "/home/da/git/pdfarranger/.venv/lib64/python3.11/site-packages/pdfarranger/metadata.py", line 85, in _safeiter
    yield next(it)
          ^^^^^^^^
  File "<frozen _collections_abc>", line 861, in __iter__
  File "/usr/lib64/python3.11/site-packages/pikepdf/models/metadata.py", line 308, in wrapper
    return fn(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/site-packages/pikepdf/models/metadata.py", line 706, in __getitem__
    raise KeyError(key) from None
KeyError: 'Zlkjsz.mJz939nwqGmtr9osNNy9z8lt-Jyt2God-QyPj9ytj.ywiPo9eQmdmKn9uPnMmTma'
```

I don't have much time currently but it looks like a but to me...

**Comment by jeromerobert** [comment]
@WB-Student May I also have the PDF file (jeromerobert at gmx.com) ?

**Comment by WB-Student** [comment]
Hi @jeromerobert,

Absolutely. I just sent them to you via email. 

Thank you.

**Comment by dreua** [comment]
@jeromerobert The workaround looks good to me, thanks! But I believe that is actually a bug in pikepdf/qpdf don't you think? I guess we still want that workaround until it is fixed and the fix is trickled through to all the distributions.

@WB-Student First of all thank you for the bug report! Would it be alright if I forwarded the files to the developers of pikepdf and maybe qpdf and (even more optionally) allow them to keep it in their *private* collection of files for future reference?

**Comment by dreua** [comment]
@jbarlow would you mind to have a look at this? I get errors when calling `items()` or `values()` on the metadata object and I don't think this is our fault. Will forward you the files if approved by the owner.

**Comment by WB-Student** [comment]
Hello @dreua and @jeromerobert,

Thank you for your help on this! I am comfortable with your sharing the files with developers. As I am not the owner of the articles, I think it should be fine sharing on a small scale for private collection. 

Thank you both for your kindness, patience, and responsitivity. I am happy this will be of use to others!

Take care.

**Comment by dreua** [comment]
Also reproducible with pikepdf 8.11.1

**Comment by jbarlow83** [comment]
@dreua I'll need the PDF.

**Comment by dreua** [comment]
I already sent it to your commit email address but that might have been an outdated clone, where do you went me to send it to?

**Comment by jbarlow83** [comment]
Sorry, found it in spam.

**Comment by jbarlow83** [comment]
Should be fixed in pikepdf 8.11.2

**Comment by dreua** [comment]
Awesome, thank you! Just published a new build to flathub with pikepdf 8.11.2 and can confirm that it works flawlessly :+1:
